agent/health_endpoint.go

```diff
@@ -264,17 +264,22 @@ func (s *HTTPServer) healthServiceNodes(resp http.ResponseWriter, req *http.Requ
 // filterNonPassing is used to filter out any nodes that have check that are not passing
 func filterNonPassing(nodes structs.CheckServiceNodes) structs.CheckServiceNodes {
 	n := len(nodes)
+
+	// Make a copy of the cached nodes rather than operating on the cache directly
+	out := make([]structs.CheckServiceNode, n)
+	out = append(nodes[:0:0], nodes...)
```

Comment:
Did you mean this?

```suggestion
	out = append(out, nodes...)
```

Reply:
Actually using your way do you need the first line to alloc the slice?

Presumably this could be condensed to:

```
out := append(nodes[:0:0], nodes...)
```

Reply:
Went with option 2, didn't need that first line. Thanks!